tests/integration/test_logcollector/test_configuration/test_basic_configuration_frequency.py

```diff
@@ -0,0 +1,168 @@
+    {'log_format': 'command', 'command': f'{command}', 'frequency': '3Testing', 'valid_value': False},
+    {'log_format': 'command', 'command': f'{command}', 'frequency': '3s5m', 'valid_value': False},
+    {'log_format': 'command', 'command': f'{command}', 'frequency': 'Testing3', 'valid_value': False},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '3', 'valid_value': True},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '10', 'valid_value': True},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '100000', 'valid_value': True},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '3s', 'valid_value': False},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': 'Testing', 'valid_value': False},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '3Testing', 'valid_value': False},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': '3s5m', 'valid_value': False},
+    {'log_format': 'full_command', 'command': f'{command}', 'frequency': 'Testing3', 'valid_value': False},
+]
+
+problematic_values = ['3s', '3s5m', '3Testing']
+
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=parameters,
+                                           metadata=metadata)
+configuration_ids = [f"{x['LOG_FORMAT'], x['COMMAND'], x['FREQUENCY']}" for x in parameters]
+
+
+def check_configuration_frequency_valid(cfg):
+    """Check if the Wazuh module run correctly and that analyze the desired file.
+
+    Ensure logcollector is running with the specified configuration, analyzing the designate file and
+    , in case of the Wazuh server, check if the API answer for localfile configuration block coincides
+    the selected configuration.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If the "Analyzing file" callback is not generated.
+        AssertError: In case of a server instance, the API response is different that the real configuration.
+    """
+
+    log_callback = logcollector.callback_monitoring_command(cfg['log_format'], cfg['command'], prefix=prefix)
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
+
+    real_configuration = cfg.copy()
+    real_configuration.pop('valid_value')
+    if wazuh_component == 'wazuh-manager':
+        api.wait_until_api_ready()
+        api.compare_config_api_response([real_configuration], 'localfile')
+
+
+def check_configuration_frequency_invalid(cfg):
+    """Check if the Wazuh fails because a invalid frequency configuration value.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If error callback are not generated.
```

Comment:
```suggestion
    """Check if the Wazuh fails because an invalid frequency configuration value.

    Args:
        cfg (dict): Dictionary with the localfile configuration.

    Raises:
        TimeoutError: If error callbacks are not generated.
```